cranelift/codegen/src/machinst/vcode.rs

```diff
@@ -1158,8 +1158,13 @@ impl<I: VCodeInst> VCode<I> {
                 let slot = alloc.as_stack().unwrap();
                 let slot_offset = self.abi.get_spillslot_offset(slot);
                 let slot_base_to_caller_sp_offset = self.abi.slot_base_to_caller_sp_offset();
+
+                #[cfg(feature = "unwind")]
                 let caller_sp_to_cfa_offset =
                     crate::isa::unwind::systemv::caller_sp_to_cfa_offset();
+                #[cfg(not(feature = "unwind"))]
+                let caller_sp_to_cfa_offset = 0; // TODO is this right?
```

Comment:
Maybe `compute_value_labels_ranges` should be made to `return ValueLabelsRanges::default();` when the unwind feature is disabled? Except for values which just so happen to be in registers, I don't think it is possible to generate valid debuginfo for locals when the unwind table is not also emitted.

Reply:
Arguably, the register values are just as 'meaningless' in the no-unwind case as you need the unwind info to recover them (unless you are willing to disassemble code to guide your unwinder). Ultimately, the debug info production 'feature' is tightly coupled to the unwinding feature.

Reply:
I marked `compute_value_labels_ranges` as available with the `unwind` feature too and made the `Default` changes at its only callsite, I think that's clearer than having the function exist but return a meaningless default value.